src/app/api/application/uniform/route.ts

```diff
@@ -0,0 +1,131 @@
+import {
+  createApplication,
+  updateApplication,
+  checkExistingReference,
+} from "@/app/actions/sanityClient";
+import { verifyApiKey } from "../../../lib/apiKey";
+import { NextRequest, NextResponse } from "next/server";
+import {
+  validateUniformData,
+  applicationNumberValidation,
+} from "@/app/actions/uniformValidator";
+
+interface ApplicationError {
+  application: any;
+  error: string;
+}
+
+interface ApplicationResult {
+  success: string[];
+  errors: ApplicationError[];
+}
+
+/**
+ * @swagger
+ * /api/application/uniform:
+ *   put:
+ *     summary: Insert new planning application
+ *     requestBody:
+ *        required: true
+ *        content:
+ *            schema:
+ *              type: object
+ *              example:
+ *                "DCAPPL[REFVAL]": "25553/5377/HSE"
+ *                "DCAPPL[BLPU_CLASS_DESC]": "Residential, Dwellings, Semi-Detached hello"
+ *                "DCAPPL[Application Type_D]": "Householder Application"
+ *     responses:
+ *       200:
+ *         description: returns updated planning applications
+ *         content:
+ *            application/json:
+ *              schema:
+ *                type: object
+ *                example: 
+ *                  data: {
+ *                    success: [
+                        "25553/5377/HSE updated"
+                      ]}
+ *       400:
+ *         description: Invalid request body or missing required fields
+ *       401:
+ *         description: Not authorized
+ *       500:
+ *         description: An error occurred while updating the applications
+ *
+ */
+
+export async function PUT(req: NextRequest) {
+  const errors: string[] = [];
+  const success: string[] = [];
+  // Verify API key
+  const referer = req.headers.get("x-api-key");
+  const apiKey = referer as string;
+  const isValidApiKey = verifyApiKey(apiKey);
+
+  if (!isValidApiKey) {
+    return NextResponse.json("Invalid API key", {
+      status: 401,
+    });
+  }
+  const data = await req.json();
+
+  const results: ApplicationResult = { success: [], errors: [] };
+  // validate
+  const validationErrors = await validateUniformData(data as any);
+  if (validationErrors.errors.length > 0) {
+    return NextResponse.json(
+      JSON.stringify({ errors: validationErrors.errors }),
+      {
+        status: validationErrors.status,
+        headers: { "Content-Type": "application/json" },
+      },
+    );
+  }
+
+  if (!data || typeof data !== "object") {
+    return new NextResponse("Invalid request body. Expected an object.", {
+      status: 400,
+    });
+  }
+
+  const applicationData = {
+    applicationNumber: data["DCAPPL[REFVAL]"],
+    description: data["DCAPPL[BLPU_CLASS_DESC]"],
+    applicationType: data["DCAPPL[Application Type_D]"],
+    isActive: true,
+    _type: "planning-application",
+  };
+
```

Comment:
I tried to submit an application with an empty applicationNumber and it allowed me to create one. I think this needs to be required field. In the applications/uniform/route.ts PUT there is a check for this. Could we add something similar here, so that we get:
`"error": "Missing required field: applicationNumber"` in the response please 😊